node/network/availability-recovery/src/lib.rs

```diff
@@ -145,33 +145,37 @@ struct InteractionParams {
 	metrics: Metrics,
 }
 
-enum InteractionPhase {
-	RequestFromBackers(RequestFromBackersPhase),
-	RequestChunks(RequestChunksPhase),
+/// Source the availability data either by means
+/// of direct request response protocol to
+/// backers (a.k.a. fast-path), or recover from chunks.
+enum Sourcer {
+	RequestFromBackers(RequestFromBackersSourcer),
+	RequestChunks(RequestChunksSourcer),
 }
 
-/// A state of a single interaction reconstructing an available data.
-struct Interaction<S> {
+/// A stateful reconstruction of availability data in reference to
+/// a candidate hash.
+struct DataRecoveryTask<S> {
 	sender: S,
 
-	/// The parameters of the interaction.
-	params: InteractionParams,
+	/// The parameters of the recovery process.
+	params: DataRecoveryParams,
 
-	/// The phase of the interaction.
-	phase: InteractionPhase,
+	/// The sourcer to obtain the availbility data.
+	sourcer: Sourcer,
 }
 
-impl RequestFromBackersPhase {
+impl RequestFromBackersSourcer {
```

Comment:
I find `Sourcer` quite weird, certainly not helpful.

Reply:
Drop it?

Reply:
I changes this because `Phase` has a timely aspect which does not exist from my understanding of the code. I am not entirely happy with `sourcer` either, but it conveys an active source while not introducing uncommon wording (i.e. `well` would by an alt discoverability).

Reply:
Dropping is fair.

Reply:
Just source? Or if you want something more active `Receiver`?